main.go

```diff
@@ -43,33 +49,50 @@ func main() {
 		log.Fatal("All scrape endpoints are disabled. You need enable at least one endpoint.")
 	}
 
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.Path != "/" {
-			http.NotFound(w, r)
-			return
-		}
+	router := getHTTPRouter()
+
+	log.WithField("port", config.BindAddr).Info("Listening for scrapes.")
+	log.WithError(router.Run(config.BindAddr)).Fatal("Shutting down.")
+}
+
+func getHTTPRouter() *gin.Engine {
+	router := gin.Default()
+
+	rg := router.Group("/")
+	rg.GET("/liveness", func(ctx *gin.Context) {
 		log.WithFields(log.Fields{
-			"uri":    r.RequestURI,
-			"client": r.RemoteAddr,
-		}).Debug("Accessed Root endpoint")
-		http.Redirect(w, r, "/metrics", http.StatusMovedPermanently)
+			"uri":    ctx.Request.RequestURI,
+			"client": ctx.Request.RemoteAddr,
+		}).Debug("Accessed Liveness endpoint")
+		ctx.String(http.StatusNoContent, "")
 	})
-	http.HandleFunc("/liveness", func(w http.ResponseWriter, r *http.Request) {
+
+	if config.BasicAuth.Username != "" {
+		if config.BasicAuth.Password == "" {
+			log.Fatal("Must set basic-auth.password to enable basic auth.")
+		}
```

Comment:
cfg.Configuration has a `postLoadProcess` func where the semantic config validation happens. I suggest to move that there as well as it's checked before the main business logic